SCons/Node/FS.py

```diff
@@ -378,20 +378,26 @@ def _my_normcase(x):
         return x.upper()
 
 
-
 class DiskChecker:
-    def __init__(self, type, do, ignore):
-        self.type = type
-        self.do = do
-        self.ignore = ignore
-        self.func = do
+    """
+    This Class will hold various types of logic for checking if a file/dir on disk matches
```

Comment:
For a new docstring, suggest following the "convention" - single-line summary, then if more needed, blank line and longer explanation. You also triggered Sider's line-length complaint here.